plugins/cilium-cni/cilium-cni.go

```diff
@@ -280,35 +288,50 @@ func cmdDel(args *skel.CmdArgs) error {
 		return fmt.Errorf("error while starting cilium-client: %s", err)
 	}
 
-	var containerIP net.IP
-	// FIXME: We need to retrieve the IPv6 address somehow...
+	var containerIPv4, containerIPv6 net.IP
+	// FIXME: We need to retrieve the IP address somehow...
 	ns.WithNetNSPath(args.Netns, func(_ ns.NetNS) error {
 		l, err := netlink.LinkByName(args.IfName)
 		if err != nil {
 			return err
 		}
-		addrs, err := netlink.AddrList(l, netlink.FAMILY_V6)
+		v6addrs, err := netlink.AddrList(l, netlink.FAMILY_V6)
 		if err != nil {
 			return err
 		}
-		log.Debugf("IPv6 addresses found %+v\n", addrs)
+		log.Debugf("IPv6 addresses found %+v\n", v6addrs)
 
-		for _, addr := range addrs {
+		for _, addr := range v6addrs {
 			if uint8(addr.Scope) == uint8(netlink.SCOPE_UNIVERSE) {
-				containerIP = addr.IP
+				containerIPv6 = addr.IP
+				break
+			}
+		}
+
+		v4addrs, err := netlink.AddrList(l, netlink.FAMILY_V4)
+		if err != nil {
+			return err
+		}
+		log.Debugf("IPv4 addresses found %+v\n", v4addrs)
```

Comment:
Can instead store the address in the endpoint together with the netns path and retrieve it that way?